### PR TITLE
Handle null parsing and partner id of weburls in webcfg

### PR DESCRIPTION
**Pull request by BalaNaveenaNivetha**
Handle null parsing and partner id of weburls in webcfg

Signed-off-by: Bala Naveena M <balanaveena_murugesan@comcast.com>

### CODE REVIEW

File: src/rdkv/impl.c
Comment (karuna2git) [inline src/rdkv/impl.c:437]: Can this be changed to call existing function?? Ref: Line#540

ie,
```
void load_def_values()
{
     snprintf(webCfgPersist.m_rfcStatus, 16, "%s", "true");
     snprintf(webCfgPersist.m_url, 1024, "%s", DEF_WEB_URL);
     snprintf(webCfgPersist.m_teleSuplUrl, 1024, "%s", DEF_SUPL_URL);
     cJSON * pWebCfg = convertWebCfgDataToJson();
     if (pWebCfg)
     {
            char* pString = cJSON_Print(pWebCfg);
            writeToFile(pString);
            cJSON_Delete(pWebCfg);
     }
}
```

**Review by karuna2git — CHANGES_REQUESTED** [review]

File: src/rdkv/impl.c
Comment (BalaNaveenaNivetha) [reply src/rdkv/impl.c:437]: I have made the respective changes in load_def_values()

File: src/rdkv/impl.c
Comment (sadhyama) [inline src/rdkv/impl.c:56]: Hard coding server urls in opensource is not ideally proper due to security reasons, or can this be read from a local tmp file ? Please check with security team and confirm to address default urls in a better way if possible..

Cc @shilpa24balaji

Comment (BalaNaveenaNivetha) [reply src/rdkv/impl.c]: Removed the hardcoding/macro and handled accordingly.

**Review by sadhyama — APPROVED** [review]

**Review by shilpa24balaji — APPROVED** [review]

**Comment by sadhyama** [comment]
Observed that same changes are added as part of cpeabs_video-changes.patch in meta layer ,
`SUMMARY = "cpeabs- Handle patches"
HOMEPAGE = "https://github.com/xmidt-org/cpeabs"

FILESEXTRAPATHS_prepend := "${THISDIR}/"
SRC_URI += "file://cpeabs_video-changes.patch"`
which is not advisable for cpeabs as it is created as an abstraction opensource layer library to avoid device specific patches. Patch should be removed and need to merge the changes in cpeabs directly which is correct method to follow. @BalaNaveenaNivetha Please confirm this and let us know if we need to merge this PR or raise new PR once patch changes are removed from rdkv meta layer. 

FYI: @shilpa24balaji

**Comment by sadhyama** [comment]
> Observed that same changes are added as part of **cpeabs_video-changes.patch** in meta layer , ` SUMMARY = "cpeabs- Handle patches" HOMEPAGE = "https://github.com/xmidt-org/cpeabs"
> 
> FILESEXTRAPATHS_prepend := "${THISDIR}/" SRC_URI += "file://cpeabs_video-changes.patch" ` which is not advisable for cpeabs as it is created as an abstraction layer opensource library to avoid device specific patches. Patch should be removed and need to merge the changes in cpeabs directly which is the correct method to follow. @BalaNaveenaNivetha Please confirm this and let us know if we need to merge this PR or raise new PR once patch changes are removed from rdkv meta layer.
> 
> FYI: @shilpa24balaji

Response from rdkv team on above issue -

`[Karunakaran A]
You merge the PR at your will. Because we cannot wait the stable2 merge until you bring in RDKC changes to stable2. Ours is P0 issue.
So we will go with patch. Get it validated today and tomorrow and go to stable2 as soon as possible.
But as you mentioned, we also do not prefer patch. But your process demands it. So we want our PR to be merged at your convenience. If you deny also, we are okey as the patch will take care.`

Holding PR merge as it will block further commits if patch is not removed/taken to yocto and it needs further discussion with rdkb team and RM .

FYI @shilpa24balaji .

**Review by karuna2git — APPROVED** [review]

**Comment by karuna2git** [comment]
Hey @sadhyama,
The changes are in releases already and now that we are working on updating the urls for webcfg. We are planning to get this merged and removed the patch file. Please merge the changes.

**Pull request merged**
